quickwit/quickwit-config/src/storage_config.rs

```diff
@@ -39,6 +39,16 @@ pub enum StorageBackend {
     S3,
 }
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum StorageBackendFlavor {
```

Comment:
@trinity-1686a, do you think it's worth adding a "Garage" flavor that forces path-style access? I know Garage supports both path and virtual-hosted style but the latter requires some DNS support.

Reply:
it would probably be a good idea. Garage doesn't profit in any way from vhost style, so it's just additional constraints for people deploying it to have a DNS wildcard (and probably the TLS certificate that comes with it)

Reply:
I would stick to what we know well today, that is `Garage`, `MinIO`, `Gcs`, there are many unknown storages and I'm afraid that `StorageBackendFlavor::SelfHosted` will not be appropriate for all self-hosted storage.